tests/core/a-entity.test.js

```diff
@@ -337,6 +337,29 @@ suite('a-entity', function () {
       assert.equal(geometry.width, 3);
     });
 
+    test('partial updates of array properties assign by reference', function () {
+      var sourceArray = [1, 2, 3];
+      registerComponent('array-comp', {
+        schema: { arr: { type: 'array' } }
+      });
+      this.el.setAttribute('array-comp', { arr: sourceArray });
+      assert.strictEqual(this.el.getAttribute('array-comp').arr, sourceArray);
+    });
+    test('partial updates of array-type properties do not trigger update', function () {
```

Comment:
I see the issue now...I would expect added or removed elements from the array to trigger an update.